src/fidesops/task/task_resources.py

```diff
@@ -157,8 +158,7 @@ def write_execution_log(  # pylint: disable=too-many-arguments
         message: str = None,
     ) -> Any:
         """Store in application db. Return the created or written-to id field value."""
-        SessionLocal = get_db_session(config)
-        db = SessionLocal()
+        db = self.session
```

Comment:
We're using the same Session that is being used for the PrivacyRequest, not creating a new one, which reduces the number of connections we're opening.  It also has a side effect of causing other resources bound to the Session for the PrivacyRequest to get the most recent state when the ExecutionLog is saved, because we commit the Session.

Reply:
I think the main implication here is with request cancellation? ie. if `.cancelled_at` / `paused_at` / `.status` were to change, we may be able to abort the traversal entirely?

Reply:
Yes, I think this is a good example of where we can make decisions mid-traversal based on the resources's current state in the future.